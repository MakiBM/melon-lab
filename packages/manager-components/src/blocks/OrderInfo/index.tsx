import React, { StatelessComponent } from 'react';
import NumberFormat from 'react-number-format';

import styles from './styles.css';

export interface OrderInfoProps {
  lastPrice: number;
  bid: number;
  ask: number;
  tokens: {
    [key: string]: {
      name: string;
      balance: number;
    };
  };
}

const OrderInfo: StatelessComponent<OrderInfoProps> = ({
  lastPrice,
  bid,
  ask,
  tokens,
}) => {
  return (
    <div className="order-info">
      <style jsx>{styles}</style>
      <div className="order-info__prices">
        <div className="order-info__last-price">
          <NumberFormat
            value={lastPrice}
            decimalScale={4}
            fixedDecimalScale={true}
            displayType={'text'}
            thousandSeparator={true}
          />
          <span className="order-info__price-desc">Last Price</span>
        </div>
        <div className="order-info__bid">
          <NumberFormat
            value={bid}
            decimalScale={4}
            fixedDecimalScale={true}
            displayType={'text'}
            thousandSeparator={true}
          />
          <span className="order-info__price-desc">Bid</span>
        </div>
        <div className="order-info__ask">
          <NumberFormat
            value={ask}
            decimalScale={4}
            fixedDecimalScale={true}
            displayType={'text'}
            thousandSeparator={true}
          />
          <span className="order-info__price-desc">Ask</span>
        </div>
      </div>
      <div className="order-info__balances">
        <div className="order-info__balance">
          <span className="order-info__balance-desc">
            {tokens.baseToken.name}:
          </span>{' '}
          <NumberFormat
            value={tokens.baseToken.balance}
            decimalScale={4}
            fixedDecimalScale={true}
            displayType={'text'}
            thousandSeparator={true}
          />
        </div>
        <div className="order-info__balance">
          <span className="order-info__balance-desc">
            {tokens.quoteToken.name}:
          </span>{' '}
          <NumberFormat
            value={tokens.quoteToken.balance}
            decimalScale={4}
            fixedDecimalScale={true}
            displayType={'text'}
            thousandSeparator={true}
          />
        </div>
      </div>
    </div>
  );
};

export default OrderInfo;
