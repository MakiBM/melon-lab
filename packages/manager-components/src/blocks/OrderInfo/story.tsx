import { storiesOf } from '@storybook/react';
import React from 'react';
import OrderInfo from './index';

const data = {
  lastPrice: 8.125,
  bid: 8.125,
  ask: 8.125,
  balances: [
    {
      name: 'ETH-T',
      value: 8.125,
    },
    {
      name: 'MLN-T',
      value: 8.125,
    },
  ],
};

storiesOf('Blocks|OrderInfo', module).add('Default', () => {
  return <OrderInfo {...data} />;
});
