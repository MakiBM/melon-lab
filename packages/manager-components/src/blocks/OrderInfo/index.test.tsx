import React from 'react';
import OrderInfo from './index';

describe('OrderInfo', () => {
  const data = {
    lastPrice: 8.125,
    bid: 8.125,
    ask: 8.125,
    balances: [
      {
        name: 'ETH-T',
        value: 8.125,
      },
      {
        name: 'MLN-T',
        value: 8.125,
      },
    ],
  };
  const defaultElement = <OrderInfo {...data} />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be selectable by class "order-info"', () => {
    expect(shallow(defaultElement).is('.order-info')).toBe(true);
  });

  it('should mount in a full DOM', () => {
    expect(mount(defaultElement).find('.order-info').length).toBe(1);
  });

  xit('should render to static HTML', () => {
    expect(render(defaultElement).text()).toEqual('Default order info');
  });
});
