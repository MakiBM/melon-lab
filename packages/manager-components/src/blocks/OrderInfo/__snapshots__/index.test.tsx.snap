// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderInfo should render correctly 1`] = `
<div
  className="jsx-__scopedHash order-info"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <div
    className="jsx-__scopedHash order-info__prices"
  >
    <div
      className="jsx-__scopedHash order-info__last-price"
    >
      8.125
       
      <span
        className="jsx-__scopedHash order-info__price-desc"
      >
        Last Price
      </span>
    </div>
    <div
      className="jsx-__scopedHash order-info__bid"
    >
      8.125
       
      <span
        className="jsx-__scopedHash order-info__price-desc"
      >
        Bid
      </span>
    </div>
    <div
      className="jsx-__scopedHash order-info__ask"
    >
      8.125
       
      <span
        className="jsx-__scopedHash order-info__price-desc"
      >
        Ask
      </span>
    </div>
  </div>
  <div
    className="jsx-__scopedHash order-info__balances"
  >
    <Balanaces
      items={
        Array [
          Object {
            "name": "ETH-T",
            "value": 8.125,
          },
          Object {
            "name": "MLN-T",
            "value": 8.125,
          },
        ]
      }
    />
  </div>
</div>
`;
